device/CMakeLists.txt

```diff
@@ -64,7 +63,7 @@ target_link_libraries(
     PRIVATE
         umd::Common
         umd::Firmware
-        hwloc
+        numa
```

Comment:
@blozano-tt does having this lib in PRIVATE section mean this will get statically linked, and thus not needed when using libdevice, just when building?

Reply:
No, PRIVATE means that -lnuma linker flag will not be transitively passed to consumers of libdevice.so

It makes sense for a dependency to be private when it is not part of the public API interface, but is instead a private implementation detail. In other words, UMD needs to link libnuma, but UMD consumers don’t have to directly link it.

Reply:
Right, so this solves a problem which you mentioned on that chat thread, or I understood that wrongly. libnuma is not needed when you want to USE umd, but it is needed when you want to build it.

Reply:
> libnuma is not needed when you want to USE umd

Not entirely true. It is not needed during link stage of compilation (between tt_metal and tt_umd). It is needed at runtime, because it is dynamically linked by UMD.

Reply:
My explanation may not have been great. 

I may have made it confusing between:
<img width="662" alt="image" src="https://github.com/user-attachments/assets/416d5a50-dac7-4b16-ab34-216732f00f7e" />

And, when the loader `ld` dynamically loads / links the library ??

Sorry, I never really polished my semantics here.